### DISCUSSION TITLE
Any way to change Input component cursor shape?

### DISCUSSION THREAD

**Discussion by dangooddd** [ArthurSonzogni/FTXUI#955] [Q&A]
Currently i can not find any information about this. As i can see, input inheritely uses focusCursorBarBlinking and focusCursorBlockBlinking. If there no way, maybe add this to InputOption?

**Comment by ArthurSonzogni** [comment]
Hello @dangooddd 

> Currently i can not find any information about this. As i can see, input inheritely uses focusCursorBarBlinking and focusCursorBlockBlinking. If there no way, maybe add this to InputOption?

We are using different shape depending on the `insert` mode.
https://github.com/ArthurSonzogni/FTXUI/blob/58ff448e768db8dffe2c79ab37e6afa66be72e0b/src/ftxui/component/input.cpp#L101-L103

A new InputOption would conflict with it?

Maybe users who need to force a certain shape could use:
```
auto input = Input(text);
auto input_wrapped = Renderer(input, [](bool focused) {
  if (focused) {
    return input->Render() | focusCursorBlockBlinking
  else {
    return input->Render();
  }
})
```
